src/neomaril_codex/datasources.py

```diff
@@ -0,0 +1,468 @@
+        dataset_hash : str
+            Name given to the datasource, should be not null, case sensitive, have between 1 and 64 characters,
+            that could be alphanumeric including accentuation (for example: 'é', à', 'ç','ñ') and space;
+        origin : Optional[str]
+            Name of the group where the datasource will be inserted.
+        
+        Returns
+        ----------
+        NeomarilDataset
+            A NeomarilDataset with the identifier as dataset_hash.
+            
+        Example
+        ----------
+        >>> dataset = datasource.get_dataset(dataset_hash='D589654eb26c4377b0df646e7a5675fa3c7d49575e03400b940dd5363006fc3a')
+        """
+        datasources = self.list_datasets(origin=origin)
+        
+        for datasource in datasources:
+            if dataset_hash == datasource.get('Id'):
+                return NeomarilDataset(
+                    dataset_hash=datasource.get('Id'),
+                    dataset_name=datasource.get('Name'),
+                    datasource_name=self.datasource_name,
+                    group=self.group,
+                    login=self.credentials[0],
+                    password=self.credentials[1],
+                    url=self.base_url
+                )
+        raise DatasetNotFoundError('Dataset hash not found!')
+
+class NeomarilDataset(BaseNeomaril):
+    """
+    Class to do operaterations in a dataset.
+
+    Attributes
+    ----------
+    login : str
+        Login for authenticating with the client.
+        You can also use the env variable NEOMARIL_USER to set this
+    password : str
+        Password for authenticating with the client.
+        You can also use the env variable NEOMARIL_PASSWORD to set this
+    url : str
+        URL to Neomaril Server.
+        Default value is https://neomaril.staging.datarisk.net,
+        use it to test your deployment first before changing to production.
+        You can also use the env variable NEOMARIL_URL to set this
+    dataset_hash : str
+        The hash that identify the dataset
+    dataset_name : str
+        The dataset defined name
+    datasource_name : str
+        Name given to the datasource, should be not null, case sensitive, have between 1 and 64 characters,
+        that could be alphanumeric including accentuation (for example: 'é', à', 'ç','ñ') and space;
+    group : str
+        Name of the group where the datasource will be inserted.
+    """
+    def __init__(self, *, dataset_hash : str, dataset_name : str, datasource_name : str, group : str, login : str, password : str, url : str) -> None:
+        super().__init__(login, password, url)
+        self.group = group
+        self.dataset_hash = dataset_hash
+        self.dataset_name = dataset_name
+        self.datasource_name = datasource_name
+
+    def get_status(self):
+        """
+        Get dataset status.
+        
+        Returns
+        ----------
+        dict
+            {
+                status : 'Succeeded',
```

Comment:
Maybe you should put the 3 possible status type

Reply:
I did with Success and Fasiled
When not found, i created an raise error

Reply:
And when it is `Running`? Or it doesn't happens to dataset?